sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RedirectStrategy.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Set;
+
+/**
+ * The interface for determining the redirect strategy used in {@link RedirectPolicy}.
+ */
+public interface RedirectStrategy {
+    ClientLogger logger = new ClientLogger(RedirectStrategy.class);
```

Comment:
nit: This should be removed as this will be a public constant on the interface. Another solution for this would be making `RedirectStrategy` an abstract class.